opal/corelib/array.rb

```diff
@@ -7,8 +7,18 @@ class Array < `Array`
   # Mark all javascript arrays as being valid ruby arrays
   `def.$$is_array = true`
 
+  %x{
+    function toArraySubclass(obj, klass) {
+      if (klass.$$name === 'Array') {
```

Comment:
this test looks a bit weak, any way to make it more reliable?

I'm thinking of stuff like:

```rb
module Mylib
  class Array < ::Array
  end
end
```